### PR TITLE
Theming the search page when no results found.

### LINKED ISSUE #3497: no search results found on site search should match copy and design of 404 no search results found
@sbsmith86 not sure if you want to keep the No Search Results Copy in http://staging.beta.dosomething.org/admin/config/dosomething/dosomething_notfound or move it somewhere else, but it'd be great to still be able to set that and have it update in both locations

### PR DESCRIPTION
**Pull request by sbsmith86**
This should handle the theming of the search page when there are no results. Had to do some digging to get to the bottom of the default no results functionality. Fixes #3497 

@aaronschachter

### CODE REVIEW

File: lib/modules/dosomething/dosomething_notfound/dosomething_notfound.admin.inc [hunk 1]
```diff
@@ -19,12 +19,6 @@ function dosomething_notfound_admin_config_form($form, &$form_state) {
     '#default_value' => variable_get('dosomething_notfound_copy_grants'),
     '#required' => TRUE,
     );
-  $form['dosomething_notfound_copy_no_results'] = array(
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_notfound/dosomething_notfound.admin.inc:22]: We should add an `hook_update_N` function into `dosomething_notfound.install` to `variable_del` this, to keep the variable table clean.

File: lib/themes/dosomething/paraneue_dosomething/includes/preprocess.inc [hunk 2]
```diff
@@ -202,6 +202,39 @@ function paraneue_dosomething_preprocess_page(&$vars) {
...
+   */
+  if (isset($vars['theme_hook_suggestions'][0])) {
+    if ($vars['theme_hook_suggestions'][0] == 'page__search' ) {
+      // If there are no results.
+      if (empty($vars['page']['content']['system_main']['search_results']['search_results']['#results'])) {
+        // Set copy.
+        $vars['noresults_copy'] = variable_get('dosomething_search_copy_no_results', 'No results found');
+
+        // Get recommended campaigns and theme them.
+        $rec_nids = dosomething_campaign_get_recommended_campaign_nids(NULL, NULL, 3);
```
Comment (aaronschachter) [inline lib/themes/dosomething/paraneue_dosomething/includes/preprocess.inc]: This all looks very similar to https://github.com/DoSomething/dosomething/blob/dev/lib/themes/dosomething/paraneue_dosomething/includes/preprocess.inc#L519 ... can we create a helper function in helpers.inc to DRY between this and the preprocess notfound?

**Comment by aaronschachter** [comment]
Code looks good.. if you have the time it'd be worth it to DRY for the Recommended Campaigns gallery tho..

**Comment by sbsmith86** [comment]
@aaronschachter I had that thought as well, actually. Yea handle the recommended campaigns gallery.

**Comment by sbsmith86** [comment]
@aaronschachter made those updates. Let me know how that update hook looks to you, i've never written one before.

File: lib/modules/dosomething/dosomething_notfound/dosomething_notfound.install [hunk 3]
```diff
@@ -1,27 +1,34 @@
...
-  // Sets the Drupal 404 URL 
+  // Sets the Drupal 404 URL
   variable_set('site_404', 'notfound');
 
 }
+
+/**
+ * Implements hook_update_N()
+ */
+function dosomethig_notfound_update_7100() {
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_notfound/dosomething_notfound.install]: I think we're better of starting at 7001.  Also, spelling error.

Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_notfound/dosomething_notfound.install]: Not sure if you have, but try running `drush updb` to make sure it runs..

File: lib/themes/dosomething/paraneue_dosomething/includes/preprocess.inc [hunk 4]
```diff
@@ -202,6 +202,21 @@ function paraneue_dosomething_preprocess_page(&$vars) {
...
+   * Search page.
+   */
+  if (isset($vars['theme_hook_suggestions'][0])) {
+    if ($vars['theme_hook_suggestions'][0] == 'page__search' ) {
+      // If there are no results.
+      if (empty($vars['page']['content']['system_main']['search_results']['search_results']['#results'])) {
+        // Set copy.
+        $vars['noresults_copy'] = variable_get('dosomething_search_copy_no_results', 'No results found');
+        // Recommended campaigns.
+        $vars['recommended_campaigns_gallery'] = paraneue_dosomething_get_recommended_campaign_gallery(NULL, NULL, 3);
```
Comment (aaronschachter) [inline lib/themes/dosomething/paraneue_dosomething/includes/preprocess.inc:216]: This rules

**Comment by aaronschachter** [comment]
Just the changes to the `hook_update` and then good to ship!  Great job value added :+1:

**Comment by sbsmith86** [comment]
@aaronschachter Fixed! :hammer:

File: lib/themes/dosomething/paraneue_dosomething/includes/preprocess.inc [hunk 5]
```diff
@@ -517,24 +532,13 @@ function paraneue_dosomething_preprocess_user_profile(&$vars) {
...
-    }
-    $vars['campaign_results'] = paraneue_dosomething_get_gallery($gallery_items, 'triad');
-  }
-  else if (isset($vars['raw_search_results'])){
+  if (isset($vars['raw_search_results']) && count($vars['raw_search_results']) != 0) {
     // Theme the set of results as duo gallery.
     $vars['search_results'] = paraneue_dosomething_get_search_gallery($vars['raw_search_results']);
   }
+  else {
+    $vars['campaign_results'] = paraneue_dosomething_get_recommended_campaign_gallery(NULL, NULL, 3);
```
Comment (aaronschachter) [inline lib/themes/dosomething/paraneue_dosomething/includes/preprocess.inc:540]: This rules too

**Comment by aaronschachter** [comment]
:+1:

**Pull request merged**